db/index.js

```diff
@@ -0,0 +1,434 @@
+  const extraPhotosQuery = queryPhotosString.substring(0, queryPhotosString.length - 2);
+  if (extraPhotosQuery.length > 0) {
+    pool.query(
+      `INSERT INTO photos(id, style_id, url, thumbnail_url) VALUES ${extraPhotosQuery}`,
+      (err, result) => {
+        if (err) {
+          console.error('Error executing extra photos query', err.stack);
+        }
+
+        return result;
+      }
+    );
+  }
+});
+
+// Styles CSV Handler
+
+const stylesFilename = './server/styles.csv';
+
+const stylesReader = fs.createReadStream(stylesFilename);
+const stylesRl = readline.createInterface({
+  input: stylesReader,
+  crlfdelay: Infinity,
+});
+stylesRl.on('line', (line) => {
+  handleStylesLine(line);
+});
+let queryStylesString = '';
+let countStyles = 0;
+
+let handleStylesLine = (line) => {
+  line = line.split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/);
+  for (let j = 0; j < line.length; j += 1) {
+    if (line[j][0] === '"') {
+      line[j] = line[j].substr(1);
+    }
+    if (line[j][line[j].length - 1] === '"') {
+      line[j] = line[j].substr(0, line[j].length - 1);
+    }
+    line[j] = line[j].replaceAll("'", "''");
+    if (line[j] === null) {
+      line[j] = 'null';
+    }
+  }
+  if (line[5] === 1) {
+    line[5] = true;
+  } else {
+    line[5] = false;
+  }
+
+  queryStylesString += `('${line[0]}', '${line[1]}', '${line[2]}', '${line[3]}', '${line[4]}', '${line[5]}'), `;
+  countStyles += 1;
+
+  if (countStyles === 1000) {
+    queryStylesString = queryStylesString.substring(0, queryStylesString.length - 2);
+    pool.query(
+      `INSERT INTO styles(style_id, product_id, name, sale_price, original_price, defaultstyle) VALUES ${queryStylesString}`,
+      (stylesErr, stylesResult) => {
+        if (stylesErr) {
+          console.error("Error executing styles query", stylesErr.stack);
+        }
+        return stylesResult;
+      }
+    );
+    queryStylesString = '';
+    countStyles = 0;
+  }
+};
+skusRl.on('close', () => {
+  const extraStylesQuery = queryStylesString.substring(0, queryStylesString.length - 2);
+  pool.query(
+    `INSERT INTO styles(style_id, product_id, name, sale_price, original_price, defaultstyle) VALUES ${extraStylesQuery}`,
+    (err, result) => {
+      if (err) {
+        console.error('Error executing extra query', err.stack);
+      }
+
+      return result;
+    }
+  );
+});
+
+// Feature CSV Handler
+const featurePort = () => {
+  const features = [];
+  const values = [];
+  let productFeaturesString = '';
+  const productFeaturesArray = [];
+  let countProductFeatures = 0;
+  const featuresFilename = './server/features.csv';
+
+  const featuresReader = fs.createReadStream(featuresFilename);
+  const featuresRl = readline.createInterface({
+    input: featuresReader,
+    crlfdelay: Infinity,
+  });
+  featuresRl.on('line', (line) => {
+    handleFeaturesLine(line);
+  });
+
+  let handleFeaturesLine = (line) => {
+    line = line.split(/,(?=(?:(?:[^"]*"){2})*[^"]*$)/);
```

Comment:
This regex splits on the commas, correct?

Reply:
exactly! splits it alllll up